main.go

```diff
@@ -204,12 +204,15 @@ func startPool(pool Service, lifecycle service.Lifecycle) error {
 	rotateSignals := make(chan os.Signal, 1)
 	signal.Notify(exitSignals, exitSignalList...)
 	signal.Notify(rotateSignals, rotateSignalList...)
+
+	deadline := cfg.SSH.GracefulTerminationDeadline + 5 * time.Second
```

Comment:
5 seconds plus is a very arbitrary number, we should come up with something better.